config-service/src/integrationTest/java/org/hypertrace/config/service/ConfigServiceIntegrationTest.java

```diff
@@ -1,97 +1,20 @@
-import org.hypertrace.config.service.v1.GetConfigRequest;
-import org.hypertrace.config.service.v1.GetConfigResponse;
-import org.hypertrace.config.service.v1.UpsertAllConfigsRequest;
-import org.hypertrace.config.service.v1.UpsertAllConfigsRequest.ConfigToUpsert;
-import org.hypertrace.config.service.v1.UpsertAllConfigsResponse;
-import org.hypertrace.config.service.v1.UpsertConfigRequest;
-import org.hypertrace.config.service.v1.UpsertConfigResponse;
-import org.hypertrace.core.documentstore.Collection;
-import org.hypertrace.core.documentstore.Datastore;
-import org.hypertrace.core.documentstore.DatastoreProvider;
-import org.hypertrace.core.grpcutils.client.RequestContextClientCallCredsProviderFactory;
+import org.hypertrace.config.service.v1.*;
 import org.hypertrace.core.grpcutils.context.RequestContext;
-import org.hypertrace.core.serviceframework.IntegrationTestServerUtil;
-import org.junit.jupiter.api.AfterAll;
-import org.junit.jupiter.api.AfterEach;
-import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
 
-/** Integration test for ConfigService */
-public class ConfigServiceIntegrationTest {
-
-  private static final String RESOURCE_NAME_1 = "foo1";
-  private static final String RESOURCE_NAME_2 = "foo2";
-  private static final String RESOURCE_NAMESPACE_1 = "bar1";
-  private static final String RESOURCE_NAMESPACE_2 = "bar2";
-  private static final String TENANT_1 = "tenant1";
-  private static final String TENANT_2 = "tenant2";
-  private static final String CONTEXT_1 = "ctx1";
-  private static final String CONTEXT_2 = "ctx2";
-  private static final String DATA_STORE_COLLECTION = "configurations";
-  private static final Collection CONFIGURATIONS_COLLECTION = getConfigurationsCollection();
-
-  private static ConfigServiceGrpc.ConfigServiceBlockingStub configServiceBlockingStub;
-  private static Value config1;
-  private static Value config2;
-  private static Value config3;
-
-  @BeforeAll
-  public static void setup() throws IOException {
-    System.out.println("Starting Config Service E2E Test");
-    IntegrationTestServerUtil.startServices(new String[] {"config-service"});
-
-    Channel channel = ManagedChannelBuilder.forAddress("localhost", 50101).usePlaintext().build();
-
-    configServiceBlockingStub =
-        ConfigServiceGrpc.newBlockingStub(channel)
-            .withCallCredentials(
-                RequestContextClientCallCredsProviderFactory.getClientCallCredsProvider().get());
-
-    config1 = getConfigValue("config1.yaml");
-    config2 = getConfigValue("config2.yaml");
-    config3 = getConfigValue("config3.yaml");
-  }
-
-  @AfterAll
-  public static void teardown() {
-    IntegrationTestServerUtil.shutdownServices();
-  }
-
-  // Need to delete the collection after each test for stateless integration testing
-  @AfterEach
-  void delete() {
-    CONFIGURATIONS_COLLECTION.deleteAll();
-  }
+public class ConfigServiceIntegrationTest extends ConfigServiceIntegrationTestBase {
```

Comment:
Test base classes are an anti-pattern and an abuse of inheritance in order to re-use code without an is-a relationship. They exist in the code in some places, but let's not go out of our way to introduce them. Prefer composition.

Reply:
removed